testnet/stacks-node/src/run_loop/nakamoto.rs

```diff
@@ -0,0 +1,1023 @@
+            &burnchain_db,
+            sortdb,
+            &sn.sortition_id,
+            &sn.canonical_stacks_tip_consensus_hash,
+            &sn.canonical_stacks_tip_hash,
+        )
+        .expect("FATAL: could not query stacks DB");
+
+        if stacks_tip_affirmation_map.len() < heaviest_affirmation_map.len()
+            || stacks_tip_affirmation_map
+                .find_divergence(&heaviest_affirmation_map)
+                .is_some()
+        {
+            // the sortition affirmation map might also be inconsistent, so we'll need to fix that
+            // (i.e. the underlying sortitions) before we can fix the stacks fork
+            if sortition_tip_affirmation_map.len() < heaviest_affirmation_map.len()
+                || sortition_tip_affirmation_map
+                    .find_divergence(&heaviest_affirmation_map)
+                    .is_some()
+            {
+                debug!("Drive burn block processing: possible PoX reorg (sortition tip: {}, heaviest: {})", &sortition_tip_affirmation_map, &heaviest_affirmation_map);
+                globals.coord().announce_new_burn_block();
+            } else if highest_sn.block_height == sn.block_height
+                && sn.block_height == canonical_burnchain_tip.block_height
+            {
+                // need to force an affirmation reorg because there will be no more burn block
+                // announcements.
+                debug!("Drive burn block processing: possible PoX reorg (sortition tip: {}, heaviest: {}, burn height {})", &sortition_tip_affirmation_map, &heaviest_affirmation_map, sn.block_height);
+                globals.coord().announce_new_burn_block();
+            }
+
+            debug!(
+                "Drive stacks block processing: possible PoX reorg (stacks tip: {}, heaviest: {})",
+                &stacks_tip_affirmation_map, &heaviest_affirmation_map
+            );
+            globals.coord().announce_new_stacks_block();
+        } else {
+            debug!(
+                "Drive stacks block processing: no need (stacks tip: {}, heaviest: {})",
+                &stacks_tip_affirmation_map, &heaviest_affirmation_map
+            );
+
+            // announce a new stacks block to force the chains coordinator
+            //  to wake up anyways. this isn't free, so we have to make sure
+            //  the chain-liveness thread doesn't wake up too often
+            globals.coord().announce_new_stacks_block();
+        }
+
+        *last_stacks_pox_reorg_recover_time = get_epoch_time_secs().into();
+    }
+
+    /// Wake up and drive sortition processing if there's been a PoX reorg.
+    /// Be careful not to saturate calls to announce new burn blocks, because that will disable
+    /// mining (which would prevent a miner attempting to fix a hidden PoX anchor block from making
+    /// progress).
+    ///
+    /// only call if no in ibd
+    fn drive_pox_reorg_burn_block_processing(
```

Comment:
Same; this can be simplified

Reply:
I removed the chain liveness thread (and its functions) for now.